osf/migrations/0113_auto_20180615_1308.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-06-15 18:08
+from __future__ import unicode_literals
+
+import logging
+from django.db import migrations
+
+from osf.models import PreprintProvider
+
+logger = logging.getLogger(__file__)
+
+
+PREPRINT_DOI_NAMESPACE = {
+    'ecsarxiv': '10.1149'
+}
+
+def add_doi_prefix(*args, **kwargs):
+    for key, value in PREPRINT_DOI_NAMESPACE.iteritems():
+        provider = PreprintProvider.objects.filter(_id=key)
+        if not provider.exists():
+            logger.info('Could not find provider with _id {}, skipping for now...'.format(key))
+            continue
+        provider = provider.get()
+        provider.doi_prefix = value
+        provider.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0112_merge_20180614_1454'),
+    ]
+
+    operations = [
+        migrations.RunPython(add_doi_prefix)
```

Comment:
How does django react to not having a reverse migration specified? I don't think a reverse migration is really needed but just don't want django to not be able to reverse beyond this one if needbe!

Reply:
Ah, good catch. 
```
Operations to perform:
  Target specific migration: 0112_merge_20180614_1454, from osf
Running migrations:
  Rendering model states... DONE
  Unapplying osf.0113_auto_20180615_1308...Traceback (most recent call last):
  File "manage.py", line 21, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/db/migrations/executor.py", line 119, in migrate
    state = self._migrate_all_backwards(plan, full_plan, fake=fake)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/db/migrations/executor.py", line 194, in _migrate_all_backwards
    self.unapply_migration(states[migration], migration, fake=fake)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/db/migrations/executor.py", line 264, in unapply_migration
    state = migration.unapply(state, schema_editor)
  File "/Users/caseyrollins/miniconda/envs/osf/lib/python2.7/site-packages/django/db/migrations/migration.py", line 152, in unapply
    raise IrreversibleError("Operation %s in %s is not reversible" % (operation, self))
django.db.migrations.exceptions.IrreversibleError: Operation <RunPython <function add_doi_prefix at 0x10d115c80>> in osf.0113_auto_20180615_1308 is not reversible
```

I'll reset to blank in the reverse.